src/platform/plugins/shared/data/common/search/search_source/types.ts

```diff
@@ -80,6 +80,13 @@ export interface SearchSourceFields {
    * {@link Filter}
    */
   filter?: Filter[] | Filter | (() => Filter[] | Filter | undefined);
+  /**
+   * Filters that should not trigger highlighting.
+   * These filters will be included in the search query for document retrieval,
+   * but excluded from the highlight_query parameter in Elasticsearch.
+   * {@link Filter}
+   */
+  nonHighlightingFilters?: Filter[] | Filter | (() => Filter[] | Filter | undefined);
```

Comment:
If possible, I'd like to avoid copying this tech debt into the new field:

```suggestion
  nonHighlightingFilters?: Filter[];
```